node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -242,19 +265,64 @@ case class P2PClientActor(
 
         currentPeerMsgHandlerRecv =
           currentPeerMsgHandlerRecv.connect(P2PClient(self, peer))
+
         context.become(awaitNetworkRequest(peerConnection, unalignedBytes))
-        unalignedBytes
+
+        Some(unalignedBytes)
 
       case closeCmd @ (Tcp.ConfirmedClosed | Tcp.Closed | Tcp.Aborted |
           Tcp.PeerClosed) =>
-        logger.info(s"We've been disconnected by $peer command=${closeCmd}")
-        //tell our peer message handler we are disconnecting
-        val newPeerMsgRecv = currentPeerMsgHandlerRecv.disconnect()
+        logger.info(
+          s"We've been disconnected by $peer command=${closeCmd} state=${currentPeerMsgHandlerRecv.state}")
+
+        currentPeerMsgHandlerRecv.state match {
+          case _: PeerMessageReceiverState.Initializing =>
```

Comment:
It looks like over-engineering. If we watch a peer connection and it gets terminated, all we need to do is just start from the scratch.